### DISCUSSION TITLE
markdown to pdf: How to include the source oder destination filename to the left in the footer?

### DISCUSSION THREAD

**Discussion by merlinuwe** [jgm/pandoc#9776] [Q&A]
I have this YAML front matter in a markdown file:

```
---
aliases:
  - Testfile
  - Obsidian
  - Enhancing
  - Export
  - plugin
tags:
  - plugin
  - Obsidian
  - miktex
  - enhancing
  - export
filename: Test table and image
sourcefile: abcdefg
erstellt: Sonntag, 19. Mai 2024, 17:00:32 Uhr
geändert: Sonntag, 19. Mai 2024, 17:30:38 Uhr
block-headings: true
dateiname: MeinDateiName.md
header-includes: |
  \usepackage{fancyhdr}
  \usepackage{hyphenat}
  \usepackage[ngerman]{babel}
  \usepackage{lastpage}
  \usepackage{caption}
  \captionsetup[table]{position=bottom}
  \pagestyle{fancy}
  \fancyhead[CO,CE]{irgendwas}
  \fancyfoot[CO,CE]{Uwe}
  \fancyfoot[L]{\jobname}
  \fancyfoot[L]{$sourcefile$}
  \fancyfoot[LE,RO]{Seite \thepage\ von \pageref{LastPage}}
  \cfoot{\thepage\ von \pageref{LastPage}}

---
```

But I only get this footer content

![image](https://github.com/jgm/pandoc/assets/4138869/1d8f9885-bb6b-4a0a-af5c-cf02bbbfd8b4)

when I convert the markdown file to pdf.

How can I get the filename of the source file or the destination file into the left side of the footer?

---

The parameters of my plugin:

Parameter `-V geometry:"top=2cm, bottom=1.5cm, left=2cm, right=1in",a4paper -f ${fromFormat} --resource-path="${currentDir}" --resource-path="${attachmentFolderPath}" --lua-filter="${luaDir}/pdf.lua" ${ options.textemplate ? `--resource-path="${pluginDir}/textemplate" --template="${options.textemplate}"` : ` ` } -o "${outputPath}" -t pdf`

Additional parameter `--pdf-engine=pdflatex`

**Comment by jgm** [comment]
The `$sourcefile$` type interpolation will only work in a template.  So, you can use a custom template instead of a header-includes. That's the easiest approach.
